File: Resolução em python/ex3.py
def calcular(peso, altura):
  imc = peso / (altura**2)
  return imc


def classificar(imc):
  if imc < 18.5:
    return "Abaixo do peso"
  elif imc < 25.0:
    return "Normal"
  elif imc < 30.0:
    return "Excesso de peso"
  elif imc < 35.0:
    return "Obesidade Grau I"
  elif imc < 40.0:
    return "Obesidade Grau II"
  else:
    return "Obesidade Grau III"


peso = float(input("Informe o peso (em kg): "))
altura = float(input("Informe a altura (em metros): "))

imc = calcular(peso, altura)
situacao = classificar(imc)

print("O seu IMC é {:.2f}".format(imc))
print("Situação: {}".format(situacao))

File: Resolução em C/ex3.c
int ex3() {
  float peso, altura, IMC;

  printf("Informe seu peso: ");
  scanf("%f", &peso);

  printf("Informe sua altura: ");
  scanf("%f", &altura);

  IMC = peso / (altura * altura);

  if (IMC < 18.5) {
    printf("Situação: Abaixo do peso\n");
  } else if (IMC >= 18.5 && IMC < 24.9) {
    printf("Situação: Normal\n");
  } else if (IMC >= 25 && IMC < 29.9) {
    printf("Situação: Excesso de peso\n");
  } else if (IMC >= 30 && IMC < 34.9) {
    printf("Situação: Obesidade Leve (Grau I)\n");
  } else if (IMC >= 35 && IMC < 39.9) {
    printf("Situação: Obesidade Severa (Grau II)\n");
  } else {
    printf("Situação: Obesidade Mórbida (Grau III)\n");
  }

  return 0;
}